### DISCUSSION TITLE
Why do some movies not show up in the frontend?

### DISCUSSION THREAD

**Discussion by othmar52** [jellyfin/jellyfin-web#6355] [Q&A] [answered]
Correctly identified movies, that are visible in the metadata manger are not available in the jellyfin frontend.  
for example:   

![image](https://github.com/user-attachments/assets/0d09db7a-c256-4d7a-9c8f-bb9b5a1d5ccf)  

no results when using the search. this item does not show up on actor pages as well.  

But it is correctly identified & fetched within the metadata manager:  
![image](https://github.com/user-attachments/assets/6738f0bb-ecd7-42f8-96ba-7a40b869312e)  

As soon as i click "Save" within the Metadata-Manager everything works like expected.  
I identified about 20-30 movies that needs this tedious procedure.  

I already considered to debug the API calls for the save action for beeing able to write a script that executes this save for every item in my collection.  
Are there any other ways?  
Or do you have an idea why this strange scenario even exists?  
most of my collection is available in jellyfin's frontend. but i have no idea how much items does not show up.  

some additional info:  
library type is movies only  
library has `.NFO`  activated
every movie has its own directory  
directory & filename is not that important because every movie gets an automated `movie.nfo` before it gets added to the library with content  
```xml
<?xml version="1.0" encoding="utf-8" standalone="yes"?>
<movie>
  <idmbid>ttXXXXXXXXXXXX</idmbid>
  <uniqueid type="imdb">ttXXXXXXXXXXXX</uniqueid>
</movie>
```
hundrets of silblings directories with the same filesystem structure are identified correctly and are available in jellyfin-web frontend  
the `movie.nfo` of the problematic items gets filled with correct metadata by jellyfin server  

Thanks for any advice!

**Comment by thornbill** [comment]
We do not provide support via GitHub discussions: https://jellyfin.org/docs/general/getting-help

**Reply by othmar52** [reply, marked as answer]
@thornbill sorry for that - it won't happen again!  
i'll post my **working solution** anyway in case anyone has the same problem...  

I made a small python script that queries jellyfins database file. for every item an search API call is performed.  
in case it is not included in the result, the item gets saved.  

this fixed availability of hundrets of items in the jellyfin frontend for me.  
its very strange that those 2 lines are even necessary, because this does not make any sense!?
```python
api_item = client.jellyfin.get_item(item_id=db_item[0])
client.jellyfin.update_item(item_id=db_item[0], data=api_item)
```

Here is the full script:  
**make sure to make database backup first!**  

```python
import sqlite3
from jellyfin_apiclient_python import JellyfinClient
from requests.exceptions import HTTPError

jellyfin_ip = '192.168.1.1'
jellyfin_username = 'admin-username'
jellyfin_password = 'admin-password'
jellyfin_db_path = "/var/lib/jellyfin/data/library.db"
client = JellyfinClient()

client.config.app('fix-missing-movies-script', '0.0.1', 'my-machine', '123456789')
client.config.data['auth.ssl'] = False
client.auth.connect_to_address(jellyfin_ip)
client.auth.login(
    'http://' + jellyfin_ip + ':8096',
    jellyfin_username,
    jellyfin_password
)

conn = sqlite3.connect(jellyfin_db_path)
cursor = conn.cursor()

cursor.execute("""
    SELECT
        PresentationUniqueKey,
        Name,
        Path
    FROM
        TypedBaseItems
    WHERE
        type = 'MediaBrowser.Controller.Entities.Movies.Movie';
""")
db_items = [row for row in cursor.fetchall()]

for db_item in db_items:
    api_items = client.jellyfin.search_media_items(term=db_item[1], media="Videos")
    found = False
    for api_item in api_items['Items']:
        if api_item['Id'] == db_item[0]:
            found = True
            break
    if found == False:
        print('UPDATING: ' + db_item[2])
        try:
            api_item = client.jellyfin.get_item(item_id=db_item[0])
            client.jellyfin.update_item(item_id=db_item[0], data=api_item)
        except HTTPError as http_err:
            print('ERROR unable to fetch & update item ' + db_item[2])
        except Exception as err:
            print('ERROR unable to fetch & update item ' + db_item[2])


conn.close()
```
